service/matching/task_queue_partition_manager.go

```diff
@@ -150,8 +209,11 @@ func (pm *taskQueuePartitionManagerImpl) GetRateLimitManager() *rateLimitManager
 // Stop does not unload the partition from matching engine. It is intended to be called by matching engine when
 // unloading the partition. For stopping and unloading a partition call unloadFromEngine instead.
 func (pm *taskQueuePartitionManagerImpl) Stop(unloadCause unloadCause) {
-	pm.versionedQueuesLock.Lock()
-	defer pm.versionedQueuesLock.Unlock()
+	pm.initCancel()
+	queue, err := pm.defaultQueueFuture.Get(context.Background())
```

Comment:
this could block? or is initialize() guaranteed to resolve the future quickly after initCtx is canceled?

Reply:
I read it over and I'm satisfied. nothing in initialize should block except wait on user data, and that will return immediately on initCancel. it may be worth a one-line comment to say that initCancel() will cause initialize to return immediately if blocked, which will resolve this future, so this shouldn't block.

the only thing to worry about would be a bug in initialize, but I think the defer will fix that.